### PR TITLE
GL core profile flag

### PR DESCRIPTION
**Pull request by kvark**
Would be really nice to have someone testing this on Windows and OSX.

### CODE REVIEW

**Comment by tomaka** [comment]
Is there really a problem with the design of `Latest`?

I really don't see the point of having the user request a core context if they can be already choose the OpenGL version. I could understand why someone would want a compatibility context (using at the same time `glBegin` and GL 4.5 functions), but that's not what your PR is.

Plus, your PR modifies `with_srgb`, which is unrelated to core/compatibility. Please open another PR for this.

**Comment by fkaa** [comment]
@tomaka Perhaps not related to the design of `Latest`, but right now (at least for cocoa) it requests the highest available version for *latest* hardware, rather than the machine it's running on, which is very specific for OS X (see [comment](https://github.com/tomaka/glutin/issues/362#issuecomment-97728684) in #362)

**Comment by kvark** [comment]
@tomaka 

> Is there really a problem with the design of Latest?

Latest as it is implemented now means the driver can give you any context it wants on GLX/WGL. For example, my linux workstation gives a 2.1 context for some reason.

> I could understand why someone would want a compatibility context (using at the same time glBegin and GL 4.5 functions), but that's not what your PR is.

It is a part of my PR. Having `with_gl_core(false)` forces the compatibility profile.

> Plus, your PR modifies with_srgb, which is unrelated to core/compatibility. Please open another PR for this.

It's such a minor and obvious change that I didn't bother. If you have any concerns about it - tell me. If not - let's not waste our time in the bureaucracy.

**Comment by tomaka** [comment]
> Latest as it is implemented now means the driver can give you any context it wants on GLX/WGL. For example, my linux workstation gives a 2.1 context for some reason.

Your PR wouldn't change anything, because the default context type is `core`.

**Comment by kvark** [comment]
> Your PR wouldn't change anything, because the default context type is core.

Neither will it break anything. Is Glutin a low-level abstraction, or a high-level? Since GLX/WGL/Cocoa explicitly differentiate between Core and Compatibility profiles, and OpenGL (of course) does, why not expose more control of it?

**Comment by tomaka** [comment]
I want to delay the change to sRGB in order to put it in version 0.1, as it's a breaking change.

For the core/compatibility thing, I guess it makes sense to offer the choice for the few people that would want this. However I'd prefer if it was `with_compatibility` instead of `with_core`.

**Comment by kvark** [comment]
@tomaka I see. How about `with_gl_profile` accepting an enum then? Not as ergonomic, but at least clearer.

**Comment by kvark** [comment]
@tomaka AppVeyor hang for some reason, but my changes are in place.

File: src/api/glx/mod.rs [hunk 1]
```diff
@@ -52,6 +53,17 @@ impl Context {
                 },
             }
 
+            if let Some(profile) = builder.gl_profile {
+                let flag = match profile {
+                    GlProfile::Compatibility =>
+                        ffi::glx_extra::CONTEXT_CORE_PROFILE_BIT_ARB,
+                    GlProfile::Core =>
+                        ffi::glx_extra::CONTEXT_COMPATIBILITY_PROFILE_BIT_ARB,
```
Comment (tomaka) [inline src/api/glx/mod.rs]: I think that's wrong

Comment (kvark) [inline src/api/glx/mod.rs]: lol, nice catch!

**Pull request merged**